// ngx-loading-x/esm5/lib/ngx-loading-x.service.js
import { __assign, __decorate, __param } from "tslib";
import { Injectable, Optional, Inject } from '@angular/core';
import { NGX_LOADING_X_TOKEN } from './ngx-loading-x-config.token';
import { DEFAULT_CONFIG } from './utils/NgxLoadingXConfig.constants';
import * as i0 from "@angular/core";
import * as i1 from "./ngx-loading-x-config.token";
var NgxLoadingXService = /** @class */ (function () {
    function NgxLoadingXService(config) {
        this.config = config;
        this.defaultConfig = __assign({}, DEFAULT_CONFIG);
        if (this.config) {
            this.defaultConfig = __assign(__assign({}, DEFAULT_CONFIG), this.config);
        }
    }
    NgxLoadingXService.prototype.getDefaultConfig = function () {
        return __assign({}, this.defaultConfig);
    };
    NgxLoadingXService.ctorParameters = function () { return [
        { type: undefined, decorators: [{ type: Optional }, { type: Inject, args: [NGX_LOADING_X_TOKEN,] }] }
    ]; };
    NgxLoadingXService.ɵprov = i0.ɵɵdefineInjectable({ factory: function NgxLoadingXService_Factory() { return new NgxLoadingXService(i0.ɵɵinject(i1.NGX_LOADING_X_TOKEN, 8)); }, token: NgxLoadingXService, providedIn: "root" });
    NgxLoadingXService = __decorate([
        Injectable({
            providedIn: 'root'
        }),
        __param(0, Optional()), __param(0, Inject(NGX_LOADING_X_TOKEN))
    ], NgxLoadingXService);
    return NgxLoadingXService;
}());
export { NgxLoadingXService };
