// ngx-loading-x/esm5/lib/ngx-loading-x.module.js
import { __decorate } from "tslib";
import { CommonModule } from '@angular/common';
import { NgModule } from '@angular/core';
import { NgxLoadingXComponent } from './ngx-loading-x.component';
import { NGX_LOADING_X_TOKEN } from './ngx-loading-x-config.token';
import { NgxLoadingXBlurDirective } from './ngx-loading-xblur.directive';
var NgxLoadingXModule = /** @class */ (function () {
    function NgxLoadingXModule() {
    }
    NgxLoadingXModule_1 = NgxLoadingXModule;
    NgxLoadingXModule.forRoot = function (ngxLoaderXConfig) {
        return {
            ngModule: NgxLoadingXModule_1,
            providers: [
                {
                    provide: NGX_LOADING_X_TOKEN,
                    useValue: ngxLoaderXConfig
                }
            ]
        };
    };
    var NgxLoadingXModule_1;
    NgxLoadingXModule = NgxLoadingXModule_1 = __decorate([
        NgModule({
            declarations: [NgxLoadingXComponent, NgxLoadingXBlurDirective],
            imports: [
                CommonModule,
            ],
            exports: [NgxLoadingXComponent, NgxLoadingXBlurDirective]
        })
    ], NgxLoadingXModule);
    return NgxLoadingXModule;
}());
export { NgxLoadingXModule };
