// ngx-loading-x/esm5/lib/ngx-loading-xblur.directive.js
import { __decorate } from "tslib";
import { Directive, Renderer2, ElementRef, Input, OnChanges } from '@angular/core';
import { NgxLoadingXService } from './ngx-loading-x.service';
var NgxLoadingXBlurDirective = /** @class */ (function () {
    function NgxLoadingXBlurDirective(renderer, elementRef, ngxLoadingXService) {
        this.renderer = renderer;
        this.elementRef = elementRef;
        this.ngxLoadingXService = ngxLoadingXService;
        this.defaultConfig = this.ngxLoadingXService.getDefaultConfig();
        this.blur = this.defaultConfig.bgBlur;
        this.show = this.defaultConfig.show;
    }
    NgxLoadingXBlurDirective.prototype.ngOnChanges = function () {
        if (this.show) {
            var blurValue = "blur(" + this.blur + "px)";
            this.renderer.setStyle(this.elementRef.nativeElement, '-webkit-filter', blurValue);
            this.renderer.setStyle(this.elementRef.nativeElement, 'filter', blurValue);
        }
        else {
            var blurValue = "none";
            this.renderer.setStyle(this.elementRef.nativeElement, '-webkit-filter', blurValue);
            this.renderer.setStyle(this.elementRef.nativeElement, 'filter', blurValue);
        }
    };
    NgxLoadingXBlurDirective.ctorParameters = function () { return [
        { type: Renderer2 },
        { type: ElementRef },
        { type: NgxLoadingXService }
    ]; };
    __decorate([
        Input()
    ], NgxLoadingXBlurDirective.prototype, "blur", void 0);
    __decorate([
        Input()
    ], NgxLoadingXBlurDirective.prototype, "show", void 0);
    NgxLoadingXBlurDirective = __decorate([
        Directive({
            selector: '[NgxLoadingXBlur]'
        })
    ], NgxLoadingXBlurDirective);
    return NgxLoadingXBlurDirective;
}());
export { NgxLoadingXBlurDirective };
