// ngx-loading-x/esm2015/lib/ngx-loading-x.service.js
import { __decorate, __param } from "tslib";
import { Injectable, Optional, Inject } from '@angular/core';
import { NGX_LOADING_X_TOKEN } from './ngx-loading-x-config.token';
import { DEFAULT_CONFIG } from './utils/NgxLoadingXConfig.constants';
import * as i0 from "@angular/core";
import * as i1 from "./ngx-loading-x-config.token";
let NgxLoadingXService = class NgxLoadingXService {
    constructor(config) {
        this.config = config;
        this.defaultConfig = Object.assign({}, DEFAULT_CONFIG);
        if (this.config) {
            this.defaultConfig = Object.assign(Object.assign({}, DEFAULT_CONFIG), this.config);
        }
    }
    getDefaultConfig() {
        return Object.assign({}, this.defaultConfig);
    }
};
NgxLoadingXService.ctorParameters = () => [
    { type: undefined, decorators: [{ type: Optional }, { type: Inject, args: [NGX_LOADING_X_TOKEN,] }] }
];
NgxLoadingXService.ɵprov = i0.ɵɵdefineInjectable({ factory: function NgxLoadingXService_Factory() { return new NgxLoadingXService(i0.ɵɵinject(i1.NGX_LOADING_X_TOKEN, 8)); }, token: NgxLoadingXService, providedIn: "root" });
NgxLoadingXService = __decorate([
    Injectable({
        providedIn: 'root'
    }),
    __param(0, Optional()), __param(0, Inject(NGX_LOADING_X_TOKEN))
], NgxLoadingXService);
export { NgxLoadingXService };
