// ngx-loading-x/esm2015/lib/ngx-loading-x.module.js
var NgxLoadingXModule_1;
import { __decorate } from "tslib";
import { CommonModule } from '@angular/common';
import { NgModule } from '@angular/core';
import { NgxLoadingXComponent } from './ngx-loading-x.component';
import { NGX_LOADING_X_TOKEN } from './ngx-loading-x-config.token';
import { NgxLoadingXBlurDirective } from './ngx-loading-xblur.directive';
let NgxLoadingXModule = NgxLoadingXModule_1 = class NgxLoadingXModule {
    static forRoot(ngxLoaderXConfig) {
        return {
            ngModule: NgxLoadingXModule_1,
            providers: [
                {
                    provide: NGX_LOADING_X_TOKEN,
                    useValue: ngxLoaderXConfig
                }
            ]
        };
    }
};
NgxLoadingXModule = NgxLoadingXModule_1 = __decorate([
    NgModule({
        declarations: [NgxLoadingXComponent, NgxLoadingXBlurDirective],
        imports: [
            CommonModule,
        ],
        exports: [NgxLoadingXComponent, NgxLoadingXBlurDirective]
    })
], NgxLoadingXModule);
export { NgxLoadingXModule };
