// ngx-loading-x/bundles/ngx-loading-x.umd.js
(function (global, factory) {
    typeof exports === 'object' && typeof module !== 'undefined' ? factory(exports, require('@angular/core'), require('@angular/platform-browser'), require('@angular/common')) :
    typeof define === 'function' && define.amd ? define('ngx-loading-x', ['exports', '@angular/core', '@angular/platform-browser', '@angular/common'], factory) :
    (global = global || self, factory(global['ngx-loading-x'] = {}, global.ng.core, global.ng.platformBrowser, global.ng.common));
}(this, (function (exports, core, platformBrowser, common) { 'use strict';

    /*! *****************************************************************************
    Copyright (c) Microsoft Corporation.

    Permission to use, copy, modify, and/or distribute this software for any
    purpose with or without fee is hereby granted.

    THE SOFTWARE IS PROVIDED "AS IS" AND THE AUTHOR DISCLAIMS ALL WARRANTIES WITH
    REGARD TO THIS SOFTWARE INCLUDING ALL IMPLIED WARRANTIES OF MERCHANTABILITY
    AND FITNESS. IN NO EVENT SHALL THE AUTHOR BE LIABLE FOR ANY SPECIAL, DIRECT,
    INDIRECT, OR CONSEQUENTIAL DAMAGES OR ANY DAMAGES WHATSOEVER RESULTING FROM
    LOSS OF USE, DATA OR PROFITS, WHETHER IN AN ACTION OF CONTRACT, NEGLIGENCE OR
    OTHER TORTIOUS ACTION, ARISING OUT OF OR IN CONNECTION WITH THE USE OR
    PERFORMANCE OF THIS SOFTWARE.
    ***************************************************************************** */
    /* global Reflect, Promise */

    var extendStatics = function(d, b) {
        extendStatics = Object.setPrototypeOf ||
            ({ __proto__: [] } instanceof Array && function (d, b) { d.__proto__ = b; }) ||
            function (d, b) { for (var p in b) if (b.hasOwnProperty(p)) d[p] = b[p]; };
        return extendStatics(d, b);
    };

    function __extends(d, b) {
        extendStatics(d, b);
        function __() { this.constructor = d; }
        d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
    }

    var __assign = function() {
        __assign = Object.assign || function __assign(t) {
            for (var s, i = 1, n = arguments.length; i < n; i++) {
                s = arguments[i];
                for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p)) t[p] = s[p];
            }
            return t;
        };
        return __assign.apply(this, arguments);
    };

    function __rest(s, e) {
        var t = {};
        for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p) && e.indexOf(p) < 0)
            t[p] = s[p];
        if (s != null && typeof Object.getOwnPropertySymbols === "function")
            for (var i = 0, p = Object.getOwnPropertySymbols(s); i < p.length; i++) {
                if (e.indexOf(p[i]) < 0 && Object.prototype.propertyIsEnumerable.call(s, p[i]))
                    t[p[i]] = s[p[i]];
            }
        return t;
    }

    function __decorate(decorators, target, key, desc) {
        var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
        if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
        else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
        return c > 3 && r && Object.defineProperty(target, key, r), r;
    }

    function __param(paramIndex, decorator) {
        return function (target, key) { decorator(target, key, paramIndex); }
    }

    function __metadata(metadataKey, metadataValue) {
        if (typeof Reflect === "object" && typeof Reflect.metadata === "function") return Reflect.metadata(metadataKey, metadataValue);
    }

    function __awaiter(thisArg, _arguments, P, generator) {
        function adopt(value) { return value instanceof P ? value : new P(function (resolve) { resolve(value); }); }
        return new (P || (P = Promise))(function (resolve, reject) {
            function fulfilled(value) { try { step(generator.next(value)); } catch (e) { reject(e); } }
            function rejected(value) { try { step(generator["throw"](value)); } catch (e) { reject(e); } }
            function step(result) { result.done ? resolve(result.value) : adopt(result.value).then(fulfilled, rejected); }
            step((generator = generator.apply(thisArg, _arguments || [])).next());
        });
    }

    function __generator(thisArg, body) {
        var _ = { label: 0, sent: function() { if (t[0] & 1) throw t[1]; return t[1]; }, trys: [], ops: [] }, f, y, t, g;
        return g = { next: verb(0), "throw": verb(1), "return": verb(2) }, typeof Symbol === "function" && (g[Symbol.iterator] = function() { return this; }), g;
        function verb(n) { return function (v) { return step([n, v]); }; }
        function step(op) {
            if (f) throw new TypeError("Generator is already executing.");
            while (_) try {
                if (f = 1, y && (t = op[0] & 2 ? y["return"] : op[0] ? y["throw"] || ((t = y["return"]) && t.call(y), 0) : y.next) && !(t = t.call(y, op[1])).done) return t;
                if (y = 0, t) op = [op[0] & 2, t.value];
                switch (op[0]) {
                    case 0: case 1: t = op; break;
                    case 4: _.label++; return { value: op[1], done: false };
                    case 5: _.label++; y = op[1]; op = [0]; continue;
                    case 7: op = _.ops.pop(); _.trys.pop(); continue;
                    default:
                        if (!(t = _.trys, t = t.length > 0 && t[t.length - 1]) && (op[0] === 6 || op[0] === 2)) { _ = 0; continue; }
                        if (op[0] === 3 && (!t || (op[1] > t[0] && op[1] < t[3]))) { _.label = op[1]; break; }
                        if (op[0] === 6 && _.label < t[1]) { _.label = t[1]; t = op; break; }
                        if (t && _.label < t[2]) { _.label = t[2]; _.ops.push(op); break; }
                        if (t[2]) _.ops.pop();
                        _.trys.pop(); continue;
                }
                op = body.call(thisArg, _);
            } catch (e) { op = [6, e]; y = 0; } finally { f = t = 0; }
            if (op[0] & 5) throw op[1]; return { value: op[0] ? op[1] : void 0, done: true };
        }
    }

    function __createBinding(o, m, k, k2) {
        if (k2 === undefined) k2 = k;
        o[k2] = m[k];
    }

    function __exportStar(m, exports) {
        for (var p in m) if (p !== "default" && !exports.hasOwnProperty(p)) exports[p] = m[p];
    }

    function __values(o) {
        var s = typeof Symbol === "function" && Symbol.iterator, m = s && o[s], i = 0;
        if (m) return m.call(o);
        if (o && typeof o.length === "number") return {
            next: function () {
                if (o && i >= o.length) o = void 0;
                return { value: o && o[i++], done: !o };
            }
        };
        throw new TypeError(s ? "Object is not iterable." : "Symbol.iterator is not defined.");
    }

    function __read(o, n) {
        var m = typeof Symbol === "function" && o[Symbol.iterator];
        if (!m) return o;
        var i = m.call(o), r, ar = [], e;
        try {
            while ((n === void 0 || n-- > 0) && !(r = i.next()).done) ar.push(r.value);
        }
        catch (error) { e = { error: error }; }
        finally {
            try {
                if (r && !r.done && (m = i["return"])) m.call(i);
            }
            finally { if (e) throw e.error; }
        }
        return ar;
    }

    function __spread() {
        for (var ar = [], i = 0; i < arguments.length; i++)
            ar = ar.concat(__read(arguments[i]));
        return ar;
    }

    function __spreadArrays() {
        for (var s = 0, i = 0, il = arguments.length; i < il; i++) s += arguments[i].length;
        for (var r = Array(s), k = 0, i = 0; i < il; i++)
            for (var a = arguments[i], j = 0, jl = a.length; j < jl; j++, k++)
                r[k] = a[j];
        return r;
    };

    function __await(v) {
        return this instanceof __await ? (this.v = v, this) : new __await(v);
    }

    function __asyncGenerator(thisArg, _arguments, generator) {
        if (!Symbol.asyncIterator) throw new TypeError("Symbol.asyncIterator is not defined.");
        var g = generator.apply(thisArg, _arguments || []), i, q = [];
        return i = {}, verb("next"), verb("throw"), verb("return"), i[Symbol.asyncIterator] = function () { return this; }, i;
        function verb(n) { if (g[n]) i[n] = function (v) { return new Promise(function (a, b) { q.push([n, v, a, b]) > 1 || resume(n, v); }); }; }
        function resume(n, v) { try { step(g[n](v)); } catch (e) { settle(q[0][3], e); } }
        function step(r) { r.value instanceof __await ? Promise.resolve(r.value.v).then(fulfill, reject) : settle(q[0][2], r); }
        function fulfill(value) { resume("next", value); }
        function reject(value) { resume("throw", value); }
        function settle(f, v) { if (f(v), q.shift(), q.length) resume(q[0][0], q[0][1]); }
    }

    function __asyncDelegator(o) {
        var i, p;
        return i = {}, verb("next"), verb("throw", function (e) { throw e; }), verb("return"), i[Symbol.iterator] = function () { return this; }, i;
        function verb(n, f) { i[n] = o[n] ? function (v) { return (p = !p) ? { value: __await(o[n](v)), done: n === "return" } : f ? f(v) : v; } : f; }
    }

    function __asyncValues(o) {
        if (!Symbol.asyncIterator) throw new TypeError("Symbol.asyncIterator is not defined.");
        var m = o[Symbol.asyncIterator], i;
        return m ? m.call(o) : (o = typeof __values === "function" ? __values(o) : o[Symbol.iterator](), i = {}, verb("next"), verb("throw"), verb("return"), i[Symbol.asyncIterator] = function () { return this; }, i);
        function verb(n) { i[n] = o[n] && function (v) { return new Promise(function (resolve, reject) { v = o[n](v), settle(resolve, reject, v.done, v.value); }); }; }
        function settle(resolve, reject, d, v) { Promise.resolve(v).then(function(v) { resolve({ value: v, done: d }); }, reject); }
    }

    function __makeTemplateObject(cooked, raw) {
        if (Object.defineProperty) { Object.defineProperty(cooked, "raw", { value: raw }); } else { cooked.raw = raw; }
        return cooked;
    };

    function __importStar(mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k in mod) if (Object.hasOwnProperty.call(mod, k)) result[k] = mod[k];
        result.default = mod;
        return result;
    }

    function __importDefault(mod) {
        return (mod && mod.__esModule) ? mod : { default: mod };
    }

    function __classPrivateFieldGet(receiver, privateMap) {
        if (!privateMap.has(receiver)) {
            throw new TypeError("attempted to get private field on non-instance");
        }
        return privateMap.get(receiver);
    }

    function __classPrivateFieldSet(receiver, privateMap, value) {
        if (!privateMap.has(receiver)) {
            throw new TypeError("attempted to set private field on non-instance");
        }
        privateMap.set(receiver, value);
        return value;
    }

    // SPinner Types

    (function (SPINNER) {
        SPINNER["xWanderingCubes"] = "x-wandering-cubes";
        SPINNER["xBallSpin"] = "x-ball-spin";
        SPINNER["xChasingDots"] = "x-chasing-dots";
        SPINNER["xElapsis"] = "x-elapsis";
        SPINNER["circleSpinner"] = "circle-spinner";
        SPINNER["threeStrings"] = "three-strings";
        SPINNER["wanderingCubes"] = "wandering-cubes";
    })(exports.SPINNER || (exports.SPINNER = {}));
    // Element Positions

    (function (POSITION) {
        POSITION["bottomCenter"] = "bottom-center";
        POSITION["bottomLeft"] = "bottom-left";
        POSITION["bottomRight"] = "bottom-right";
        POSITION["centerCenter"] = "center-center";
        POSITION["centerLeft"] = "center-left";
        POSITION["centerRight"] = "center-right";
        POSITION["topCenter"] = "top-center";
        POSITION["topLeft"] = "top-left";
        POSITION["topRight"] = "top-right";
    })(exports.POSITION || (exports.POSITION = {}));

    var NGX_LOADING_X_TOKEN = new core.InjectionToken('ngxLoadingXConfig.config');

    var DEFAULT_CONFIG = {
        show: false,
        bgBlur: 5,
        bgColor: 'rgba(40, 40, 40, 0.5)',
        bgOpacity: 5,
        bgLogoUrl: '',
        bgLogoUrlPosition: exports.POSITION.bottomRight,
        bgLogoUrlSize: 100,
        spinnerType: exports.SPINNER.wanderingCubes,
        spinnerSize: 120,
        spinnerColor: '#dd1b16',
        spinnerPosition: exports.POSITION.centerCenter,
    };
    // Spinner Configuration
    var SPINNER_CONFIG = {
        'x-wandering-cubes': {
            divs: 2,
            class: 'x-wandering-cubes'
        },
        'x-ball-spin': {
            divs: 8,
            class: 'x-ball-spin'
        },
        'x-chasing-dots': {
            divs: 2,
            class: 'x-chasing-dots'
        },
        'x-elapsis': {
            divs: 12,
            class: 'x-elapsis'
        },
        'circle-spinner': {
            divs: 12,
            class: 'circle-spinner'
        },
        'three-strings': {
            divs: 3,
            class: 'sk-three-strings'
        },
        'wandering-cubes': {
            divs: 2,
            class: 'sk-wandering-cubes'
        }
    };

    var NgxLoadingXService = /** @class */ (function () {
        function NgxLoadingXService(config) {
            this.config = config;
            this.defaultConfig = __assign({}, DEFAULT_CONFIG);
            if (this.config) {
                this.defaultConfig = __assign(__assign({}, DEFAULT_CONFIG), this.config);
            }
        }
        NgxLoadingXService.prototype.getDefaultConfig = function () {
            return __assign({}, this.defaultConfig);
        };
        NgxLoadingXService.ctorParameters = function () { return [
            { type: undefined, decorators: [{ type: core.Optional }, { type: core.Inject, args: [NGX_LOADING_X_TOKEN,] }] }
        ]; };
        NgxLoadingXService.ɵprov = core.ɵɵdefineInjectable({ factory: function NgxLoadingXService_Factory() { return new NgxLoadingXService(core.ɵɵinject(NGX_LOADING_X_TOKEN, 8)); }, token: NgxLoadingXService, providedIn: "root" });
        NgxLoadingXService = __decorate([
            core.Injectable({
                providedIn: 'root'
            }),
            __param(0, core.Optional()), __param(0, core.Inject(NGX_LOADING_X_TOKEN))
        ], NgxLoadingXService);
        return NgxLoadingXService;
    }());

    var NgxLoadingXBlurDirective = /** @class */ (function () {
        function NgxLoadingXBlurDirective(renderer, elementRef, ngxLoadingXService) {
            this.renderer = renderer;
            this.elementRef = elementRef;
            this.ngxLoadingXService = ngxLoadingXService;
            this.defaultConfig = this.ngxLoadingXService.getDefaultConfig();
            this.blur = this.defaultConfig.bgBlur;
            this.show = this.defaultConfig.show;
        }
        NgxLoadingXBlurDirective.prototype.ngOnChanges = function () {
            if (this.show) {
                var blurValue = "blur(" + this.blur + "px)";
                this.renderer.setStyle(this.elementRef.nativeElement, '-webkit-filter', blurValue);
                this.renderer.setStyle(this.elementRef.nativeElement, 'filter', blurValue);
            }
            else {
                var blurValue = "none";
                this.renderer.setStyle(this.elementRef.nativeElement, '-webkit-filter', blurValue);
                this.renderer.setStyle(this.elementRef.nativeElement, 'filter', blurValue);
            }
        };
        NgxLoadingXBlurDirective.ctorParameters = function () { return [
            { type: core.Renderer2 },
            { type: core.ElementRef },
            { type: NgxLoadingXService }
        ]; };
        __decorate([
            core.Input()
        ], NgxLoadingXBlurDirective.prototype, "blur", void 0);
        __decorate([
            core.Input()
        ], NgxLoadingXBlurDirective.prototype, "show", void 0);
        NgxLoadingXBlurDirective = __decorate([
            core.Directive({
                selector: '[NgxLoadingXBlur]'
            })
        ], NgxLoadingXBlurDirective);
        return NgxLoadingXBlurDirective;
    }());

    var NgxLoadingXComponent = /** @class */ (function () {
        function NgxLoadingXComponent(domSanitizer, ngxLoadingXService) {
            this.domSanitizer = domSanitizer;
            this.ngxLoadingXService = ngxLoadingXService;
            this.defaultConfig = this.ngxLoadingXService.getDefaultConfig();
            this.show = this.defaultConfig.show;
            this.bgBlur = this.defaultConfig.bgBlur;
            this.bgColor = this.defaultConfig.bgColor;
            this.bgOpacity = this.defaultConfig.bgOpacity;
            this.bgLogoUrl = this.defaultConfig.bgLogoUrl;
            this.bgLogoUrlPosition = this.defaultConfig.bgLogoUrlPosition;
            this.bgLogoUrlSize = this.defaultConfig.bgLogoUrlSize;
            this.spinnerType = this.defaultConfig.spinnerType;
            this.spinnerSize = this.defaultConfig.spinnerSize;
            this.spinnerColor = this.defaultConfig.spinnerColor;
            this.spinnerPosition = this.defaultConfig.spinnerPosition;
        }
        NgxLoadingXComponent.prototype.initializeSpinners = function () {
            this.spinnerDivs = Array(SPINNER_CONFIG[this.spinnerType].divs).fill(1);
            this.spinnerClass = SPINNER_CONFIG[this.spinnerType].class;
        };
        NgxLoadingXComponent.prototype.determineCenterPositions = function () {
            if (this.bgLogoUrlPosition === exports.POSITION.centerCenter && this.spinnerPosition === exports.POSITION.centerCenter) {
                this.spinnerTop = this.domSanitizer.bypassSecurityTrustStyle("calc(50% + 88px)");
                this.logoTop = this.domSanitizer.bypassSecurityTrustStyle("calc(50% - 88px)");
            }
        };
        NgxLoadingXComponent.prototype.ngOnInit = function () {
            this.trustedLogoUrl = this.domSanitizer.bypassSecurityTrustResourceUrl(this.bgLogoUrl);
            this.initializeSpinners();
            this.determineCenterPositions();
        };
        NgxLoadingXComponent.ctorParameters = function () { return [
            { type: platformBrowser.DomSanitizer },
            { type: NgxLoadingXService }
        ]; };
        __decorate([
            core.Input()
        ], NgxLoadingXComponent.prototype, "show", void 0);
        __decorate([
            core.Input()
        ], NgxLoadingXComponent.prototype, "bgBlur", void 0);
        __decorate([
            core.Input()
        ], NgxLoadingXComponent.prototype, "bgColor", void 0);
        __decorate([
            core.Input()
        ], NgxLoadingXComponent.prototype, "bgOpacity", void 0);
        __decorate([
            core.Input()
        ], NgxLoadingXComponent.prototype, "bgLogoUrl", void 0);
        __decorate([
            core.Input()
        ], NgxLoadingXComponent.prototype, "bgLogoUrlPosition", void 0);
        __decorate([
            core.Input()
        ], NgxLoadingXComponent.prototype, "bgLogoUrlSize", void 0);
        __decorate([
            core.Input()
        ], NgxLoadingXComponent.prototype, "spinnerType", void 0);
        __decorate([
            core.Input()
        ], NgxLoadingXComponent.prototype, "spinnerSize", void 0);
        __decorate([
            core.Input()
        ], NgxLoadingXComponent.prototype, "spinnerColor", void 0);
        __decorate([
            core.Input()
        ], NgxLoadingXComponent.prototype, "spinnerPosition", void 0);
        NgxLoadingXComponent = __decorate([
            core.Component({
                selector: 'ngx-loading-x',
                template: "<!-- container -->\n<div *ngIf=\"show\">\n  <div class=\"ngx-overlay\" [style.background-color]=\"bgColor\" [style.opacity]=\"bgOpacity\">\n    <!-- Logo -->\n    <img *ngIf=\"bgLogoUrl\" class=\"ngx-loading-x-logo\" [ngClass]=\"bgLogoUrlPosition\" [src]=\"trustedLogoUrl\"\n      [style.width.px]=\"bgLogoUrlSize\" [style.height.px]=\"bgLogoUrlSize\" [style.top]=\"logoTop\" />\n    <!-- Logo -->\n\n    <!-- spinner -->\n    <div class=\"ngx-foreground-spinner\" [ngClass]=\"spinnerPosition\" [style.color]=\"spinnerColor\"\n      [style.width.px]=\"spinnerSize\" [style.height.px]=\"spinnerSize\" [style.top]=\"spinnerTop\">\n      <div [class]=\"spinnerClass\">\n        <div *ngFor=\"let div of spinnerDivs\"></div>\n      </div>\n    </div>\n  </div>\n  <!-- End Spinner -->\n\n\n</div>\n\n<!-- container -->\n",
                changeDetection: core.ChangeDetectionStrategy.OnPush,
                styles: [".ngx-overlay{position:fixed;top:0;left:0;width:100%;height:100%;z-index:99998!important;cursor:progress;transition:5s ease-out}.ngx-overlay>.ngx-foreground-spinner,.ngx-overlay>.ngx-loading-x-logo{position:fixed;margin:0}.ngx-background-spinner{position:fixed;z-index:99997!important;margin:0;opacity:.6}.ngx-position-absolute,.ngx-position-absolute>.ngx-foreground-spinner,.ngx-position-absolute>.ngx-loading-x-logo{position:absolute!important}.ngx-position-absolute.ngx-overlay{z-index:99995!important}.ngx-position-absolute.ngx-background-spinner{z-index:99994!important}.top-left{top:30px;left:30px}.top-center{top:30px;left:50%;transform:translateX(-50%)}.top-right{top:30px;right:30px}.center-left{top:50%;left:30px;transform:translateY(-50%)}.center-center{top:50%;left:50%;transform:translate(-50%,-50%)}.center-right{top:50%;right:30px;transform:translateY(-50%)}.bottom-left{bottom:30px;left:30px}.bottom-center{bottom:30px;left:50%;transform:translateX(-50%)}.bottom-right{bottom:30px;right:30px}.x-ball-spin,.x-ball-spin>div{position:relative;box-sizing:border-box}.x-ball-spin{width:100%;height:100%;font-size:0}.x-ball-spin>div{display:inline-block;float:none;background-color:currentColor;border:0 solid currentColor;position:absolute;top:50%;left:50%;width:25%;height:25%;margin-top:-12.5%;margin-left:-12.5%;border-radius:100%;-webkit-animation:1s ease-in-out infinite ball-spin;animation:1s ease-in-out infinite ball-spin}.x-ball-spin>div:nth-child(1){top:5%;left:50%;-webkit-animation-delay:-1.125s;animation-delay:-1.125s}.x-ball-spin>div:nth-child(2){top:18.1801948466%;left:81.8198051534%;-webkit-animation-delay:-1.25s;animation-delay:-1.25s}.x-ball-spin>div:nth-child(3){top:50%;left:95%;-webkit-animation-delay:-1.375s;animation-delay:-1.375s}.x-ball-spin>div:nth-child(4){top:81.8198051534%;left:81.8198051534%;-webkit-animation-delay:-1.5s;animation-delay:-1.5s}.x-ball-spin>div:nth-child(5){top:94.9999999966%;left:50.0000000005%;-webkit-animation-delay:-1.625s;animation-delay:-1.625s}.x-ball-spin>div:nth-child(6){top:81.8198046966%;left:18.1801949248%;-webkit-animation-delay:-1.75s;animation-delay:-1.75s}.x-ball-spin>div:nth-child(7){top:49.9999750815%;left:5.0000051215%;-webkit-animation-delay:-1.875s;animation-delay:-1.875s}.x-ball-spin>div:nth-child(8){top:18.179464974%;left:18.1803700518%;-webkit-animation-delay:-2s;animation-delay:-2s}@-webkit-keyframes ball-spin{0%,100%{opacity:1;transform:scale(1)}20%{opacity:1}80%{opacity:0;transform:scale(0)}}@keyframes ball-spin{0%,100%{opacity:1;transform:scale(1)}20%{opacity:1}80%{opacity:0;transform:scale(0)}}.x-chasing-dots{margin:auto;width:100%;height:100%;position:absolute;text-align:center;-webkit-animation:2s linear infinite x-chasingDots-rotate;animation:2s linear infinite x-chasingDots-rotate}.x-chasing-dots>div{width:60%;height:60%;display:inline-block;position:absolute;top:0;background-color:currentColor;border-radius:100%;-webkit-animation:2s ease-in-out infinite x-chasingDots-bounce;animation:2s ease-in-out infinite x-chasingDots-bounce}.x-chasing-dots>div:nth-child(2){top:auto;bottom:0;-webkit-animation-delay:-1s;animation-delay:-1s}@-webkit-keyframes x-chasingDots-rotate{100%{transform:rotate(360deg)}}@keyframes x-chasingDots-rotate{100%{transform:rotate(360deg)}}@-webkit-keyframes x-chasingDots-bounce{0%,100%{transform:scale(0)}50%{transform:scale(1)}}@keyframes x-chasingDots-bounce{0%,100%{transform:scale(0)}50%{transform:scale(1)}}.x-elapsis{display:inline-block;position:relative;width:80px;height:80px}.x-elapsis div{position:absolute;top:33px;width:13px;height:13px;border-radius:50%;background:currentColor;-webkit-animation-timing-function:cubic-bezier(0,1,1,0);animation-timing-function:cubic-bezier(0,1,1,0)}.x-elapsis div:nth-child(1){left:8px;-webkit-animation:.6s infinite lds-ellipsis1;animation:.6s infinite lds-ellipsis1}.x-elapsis div:nth-child(2){left:8px;-webkit-animation:.6s infinite lds-ellipsis2;animation:.6s infinite lds-ellipsis2}.x-elapsis div:nth-child(3){left:32px;-webkit-animation:.6s infinite lds-ellipsis2;animation:.6s infinite lds-ellipsis2}.x-elapsis div:nth-child(4){left:56px;-webkit-animation:.6s infinite lds-ellipsis3;animation:.6s infinite lds-ellipsis3}@-webkit-keyframes lds-ellipsis1{0%{transform:scale(0)}100%{transform:scale(1)}}@keyframes lds-ellipsis1{0%{transform:scale(0)}100%{transform:scale(1)}}@-webkit-keyframes lds-ellipsis3{0%{transform:scale(1)}100%{transform:scale(0)}}@keyframes lds-ellipsis3{0%{transform:scale(1)}100%{transform:scale(0)}}@-webkit-keyframes lds-ellipsis2{0%{transform:translate(0,0)}100%{transform:translate(24px,0)}}@keyframes lds-ellipsis2{0%{transform:translate(0,0)}100%{transform:translate(24px,0)}}.circle-spinner{color:official;display:inline-block;position:relative;width:80px;height:80px}.circle-spinner div{transform-origin:40px 40px;-webkit-animation:1.2s linear infinite circle-spinner;animation:1.2s linear infinite circle-spinner}.circle-spinner div:after{content:\" \";display:block;position:absolute;top:3px;left:37px;width:6px;height:18px;border-radius:20%;background:currentColor}.circle-spinner div:nth-child(1){transform:rotate(0);-webkit-animation-delay:-1.1s;animation-delay:-1.1s}.circle-spinner div:nth-child(2){transform:rotate(30deg);-webkit-animation-delay:-1s;animation-delay:-1s}.circle-spinner div:nth-child(3){transform:rotate(60deg);-webkit-animation-delay:-.9s;animation-delay:-.9s}.circle-spinner div:nth-child(4){transform:rotate(90deg);-webkit-animation-delay:-.8s;animation-delay:-.8s}.circle-spinner div:nth-child(5){transform:rotate(120deg);-webkit-animation-delay:-.7s;animation-delay:-.7s}.circle-spinner div:nth-child(6){transform:rotate(150deg);-webkit-animation-delay:-.6s;animation-delay:-.6s}.circle-spinner div:nth-child(7){transform:rotate(180deg);-webkit-animation-delay:-.5s;animation-delay:-.5s}.circle-spinner div:nth-child(8){transform:rotate(210deg);-webkit-animation-delay:-.4s;animation-delay:-.4s}.circle-spinner div:nth-child(9){transform:rotate(240deg);-webkit-animation-delay:-.3s;animation-delay:-.3s}.circle-spinner div:nth-child(10){transform:rotate(270deg);-webkit-animation-delay:-.2s;animation-delay:-.2s}.circle-spinner div:nth-child(11){transform:rotate(300deg);-webkit-animation-delay:-.1s;animation-delay:-.1s}.circle-spinner div:nth-child(12){transform:rotate(330deg);-webkit-animation-delay:0s;animation-delay:0s}@-webkit-keyframes circle-spinner{0%{opacity:1}100%{opacity:0}}@keyframes circle-spinner{0%{opacity:1}100%{opacity:0}}.sk-three-strings{width:100%;height:100%}.sk-three-strings>div{position:absolute;box-sizing:border-box;width:100%;height:100%;border-radius:50%}.sk-three-strings>div:nth-child(1){left:0;top:0;-webkit-animation:1s linear infinite sk-threeStrings-rotateOne;animation:1s linear infinite sk-threeStrings-rotateOne;border-bottom:3px solid currentColor}.sk-three-strings>div:nth-child(2){right:0;top:0;-webkit-animation:1s linear infinite sk-threeStrings-rotateTwo;animation:1s linear infinite sk-threeStrings-rotateTwo;border-right:3px solid currentColor}.sk-three-strings>div:nth-child(3){right:0;bottom:0;-webkit-animation:1s linear infinite sk-threeStrings-rotateThree;animation:1s linear infinite sk-threeStrings-rotateThree;border-top:3px solid currentColor}@-webkit-keyframes sk-threeStrings-rotateOne{0%{transform:rotateX(35deg) rotateY(-45deg) rotateZ(0)}100%{transform:rotateX(35deg) rotateY(-45deg) rotateZ(360deg)}}@keyframes sk-threeStrings-rotateOne{0%{transform:rotateX(35deg) rotateY(-45deg) rotateZ(0)}100%{transform:rotateX(35deg) rotateY(-45deg) rotateZ(360deg)}}@-webkit-keyframes sk-threeStrings-rotateTwo{0%{transform:rotateX(50deg) rotateY(10deg) rotateZ(0)}100%{transform:rotateX(50deg) rotateY(10deg) rotateZ(360deg)}}@keyframes sk-threeStrings-rotateTwo{0%{transform:rotateX(50deg) rotateY(10deg) rotateZ(0)}100%{transform:rotateX(50deg) rotateY(10deg) rotateZ(360deg)}}@-webkit-keyframes sk-threeStrings-rotateThree{0%{transform:rotateX(35deg) rotateY(55deg) rotateZ(0)}100%{transform:rotateX(35deg) rotateY(55deg) rotateZ(360deg)}}@keyframes sk-threeStrings-rotateThree{0%{transform:rotateX(35deg) rotateY(55deg) rotateZ(0)}100%{transform:rotateX(35deg) rotateY(55deg) rotateZ(360deg)}}.sk-wandering-cubes{margin:auto;width:100%;height:100%;position:relative;text-align:center}.sk-wandering-cubes>div{background-color:currentColor;width:25%;height:25%;position:absolute;top:0;left:0;-webkit-animation:1.8s ease-in-out infinite sk-wanderingCubes-cubeMove;animation:1.8s ease-in-out infinite sk-wanderingCubes-cubeMove}.sk-wandering-cubes>div:nth-child(2){-webkit-animation-delay:-.9s;animation-delay:-.9s}@-webkit-keyframes sk-wanderingCubes-cubeMove{25%{transform:translateX(290%) rotate(-90deg) scale(.5)}50%{transform:translateX(290%) translateY(290%) rotate(-179deg)}50.1%{transform:translateX(290%) translateY(290%) rotate(-180deg)}75%{transform:translateX(0) translateY(290%) rotate(-270deg) scale(.5)}100%{transform:rotate(-360deg)}}@keyframes sk-wanderingCubes-cubeMove{25%{transform:translateX(290%) rotate(-90deg) scale(.5)}50%{transform:translateX(290%) translateY(290%) rotate(-179deg)}50.1%{transform:translateX(290%) translateY(290%) rotate(-180deg)}75%{transform:translateX(0) translateY(290%) rotate(-270deg) scale(.5)}100%{transform:rotate(-360deg)}}.x-wandering-cubes{margin:auto;width:100%;height:100%;position:relative;transform:rotateZ(45deg)}.x-wandering-cubes>div{float:left;width:50%;height:50%;position:relative;transform:scale(1.1)}.x-wandering-cubes>div::before{content:\"\";position:absolute;top:0;left:0;width:100%;height:100%;background-color:currentColor;-webkit-animation:2.4s linear infinite both sk-foldingCube-angle;animation:2.4s linear infinite both sk-foldingCube-angle;transform-origin:100% 100%}.x-wandering-cubes>div:nth-child(2){transform:scale(1.1) rotateZ(90deg)}.x-wandering-cubes>div:nth-child(3){transform:scale(1.1) rotateZ(270deg)}.x-wandering-cubes>div:nth-child(4){transform:scale(1.1) rotateZ(180deg)}.x-wandering-cubes>div:nth-child(2)::before{-webkit-animation-delay:.3s;animation-delay:.3s}.x-wandering-cubes>div:nth-child(3)::before{-webkit-animation-delay:.9s;animation-delay:.9s}.x-wandering-cubes>div:nth-child(4)::before{-webkit-animation-delay:.6s;animation-delay:.6s}@-webkit-keyframes sk-foldingCube-angle{0%,10%{transform:perspective(140px) rotateX(-180deg);opacity:0}25%,75%{transform:perspective(140px) rotateX(0);opacity:1}100%,90%{transform:perspective(140px) rotateY(180deg);opacity:0}}@keyframes sk-foldingCube-angle{0%,10%{transform:perspective(140px) rotateX(-180deg);opacity:0}25%,75%{transform:perspective(140px) rotateX(0);opacity:1}100%,90%{transform:perspective(140px) rotateY(180deg);opacity:0}}"]
            })
        ], NgxLoadingXComponent);
        return NgxLoadingXComponent;
    }());

    var NgxLoadingXModule = /** @class */ (function () {
        function NgxLoadingXModule() {
        }
        NgxLoadingXModule_1 = NgxLoadingXModule;
        NgxLoadingXModule.forRoot = function (ngxLoaderXConfig) {
            return {
                ngModule: NgxLoadingXModule_1,
                providers: [
                    {
                        provide: NGX_LOADING_X_TOKEN,
                        useValue: ngxLoaderXConfig
                    }
                ]
            };
        };
        var NgxLoadingXModule_1;
        NgxLoadingXModule = NgxLoadingXModule_1 = __decorate([
            core.NgModule({
                declarations: [NgxLoadingXComponent, NgxLoadingXBlurDirective],
                imports: [
                    common.CommonModule,
                ],
                exports: [NgxLoadingXComponent, NgxLoadingXBlurDirective]
            })
        ], NgxLoadingXModule);
        return NgxLoadingXModule;
    }());

    exports.NgxLoadingXBlurDirective = NgxLoadingXBlurDirective;
    exports.NgxLoadingXComponent = NgxLoadingXComponent;
    exports.NgxLoadingXModule = NgxLoadingXModule;
    exports.NgxLoadingXService = NgxLoadingXService;
    exports.ɵa = NGX_LOADING_X_TOKEN;

    Object.defineProperty(exports, '__esModule', { value: true });

})));
